### PR TITLE
Typeorm integration sql

### PR DESCRIPTION
**Pull request by ghost**
Adding typeorm to the project according to their documentation.
[](https://typeorm.io/)

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -2,4 +2,9 @@ TOKEN=
...
 GUILD_ID=
 SOURCEJUMP_API_KEY=
-SOURCEJUMP_API_URL=
\ No newline at end of file
+SOURCEJUMP_API_URL=
+SQL_HOST=
+SQL_USER=
+SQL_DATABASE=
+SQL_PORT=
+SQL_PW=
```
Comment (ecsr) [inline .env.example]: Can we change this to DATABASE_URL and use a MySQL connection string like `mysql://user:password@host/db` instead?

Comment (ghost) [reply .env.example]: > Can we change this to DATABASE_URL and use a MySQL connection string like `mysql://user:password@host/db` instead?

Yes typeorm provides a `url` option for this

File: src/database/db.js [hunk 2]
```diff
@@ -2,11 +2,8 @@ const typeorm = require('typeorm');
 
 const typeormConnection = new typeorm.DataSource({
     type: 'mysql',
-    host: process.env.SQL_HOST,
-    username: process.env.SQL_USER,
-    database: process.env.SQL_DATABASE,
+    url: process.env.DATABASE_URL,
     port: process.env.SQL_PORT,
```
Comment (ecsr) [inline src/database/db.js]: Port can be included in the URL as well so this can be removed.

Comment (ghost) [reply src/database/db.js]: Forgot to remove thank you!

**Review by ecsr — APPROVED** [review]

**Pull request merged**